Feature/segregazione delivery push

speratimatteo opened the pull request:
Test per segregazione delivery push.
Trattasi di suite temporanee (che a feature conclusa si potranno tranquillamente cancellare, poichè anche le API richiamate non esisteranno più).
Il loro scopo infatti, più che testare un comportamento atteso, è individuare eventuali discrepanze che potrebbero essere insorte a livello di timeline o status history con le modifiche apportate su delivery push.
I test, estraggono un certo numero di notifiche comprese tra dataA e dataB, dopodichè, per ciascuna di questa notifiche invocano l'api con versione nuova e vecchia, e mettono a confronto le stringhe json ottenute.

speratimatteo committed:
Segregazione delivery push

FIX per alcuni test tra cui [B2B-RECIPIENT_LEGALFACT_3]

speratimatteo committed:
segregazione dp

Fix per sending receipt

speratimatteo committed:
Segregazione DP

Implementazione TA for Segregazione delivery push

speratimatteo committed:
Merge branch 'develop' into feature/segregazione_delivery_push

speratimatteo committed:
Segregazione DP

Modifica al pom per execution template engine (punta a main, non più a develop)

speratimatteo committed:
Segregazione DP

Creazione suite segregazione delivery push. Aumento range copertura notifiche

speratimatteo committed:
segregazione DP

Edit property for deliveryPush base url

speratimatteo committed:
Merge branch 'develop' into feature/segregazione_delivery_push

mmancini95 commented on src/main/java/it/pagopa/pn/client/b2b/pa/service/impl/PnWebPaClientImpl.java:
```diff
@@ -14,7 +14,9 @@
-public class PnWebPaClientImpl implements IPnWebPaClient {
+//TODO: anzichè eliminare del tutto la classe, ho commentato implements IPnWebPaClient e gli @Override (tante volte dovessimo tornare a usarla)
+//public class PnWebPaClientImpl implements IPnWebPaClient {
+public class PnWebPaClientImpl {
```
Per mantenere il codice un po' più pulito, cancellerei del tutto questa classe, dato che ormai dovremmo sempre passare per il BFF

mmancini95 commented on src/test/java/it/pagopa/pn/cucumber/steps/pa/SegregazioneDeliveryPushSteps.java:
```diff
@@ -0,0 +1,213 @@
+    }
+
+    @Then("confronto gli status history ottenuti chiamando la nuova API e la vecchia")
+    public void compareStatusHistoryResults() {
```
è possibile esternalizzare in un metodo privato alcune parti di questa logica per essere poi invocate sia in compareStatusHistoryResults e compareTimelineResults? In modo da evitare duplicazione di codice

mmancini95 commented on src/test/java/it/pagopa/pn/cucumber/utils/datatestVersions/DataTestV25.java:
```diff
@@ -138,9 +138,15 @@ public static void checkTimelineElementEquality(ApplicationContext context, Stri
+                        assertThat(actual.getSendingReceipts().get(i)).as("Il sendingReceipt non dev'essere null").isNotNull();
+                        assertThat(actual.getSendingReceipts().get(i).getId()).as("L'ID del sendingReceipt non dev'essere null").isNotNull();
+                        assertThat(actual.getSendingReceipts().get(i).getSystem()).as("Il System del sendingReceipt non dev'essere null").isNotNull();
+                        //TODO 26/06/25: a seguito di recenti modifiche i campi ID e System non arrivano più nul
```
abbiamo modo di capire qual è il valore realmente atteso per questi campi?

speratimatteo committed:
Segregazione DP

Eliminazione classe PnWebPaClientImpl, sostituita in toto da PnBffPaClientImpl.
Applicazione della stessa logica usata in DataTestV25 anche sugli altri DataTest
